packages/project/src/db/props.ts

```diff
@@ -1,22 +1,73 @@
 import {
   type Tree,
   type AllUserProps,
-  UserProps,
+  type UserProp,
+  // TODO: move zod schema into project package
+  UserUserProps,
+  UserDbProps,
 } from "@webstudio-is/react-sdk";
 import { applyPatches, type Patch } from "immer";
 import { prisma } from "@webstudio-is/prisma-client";
+import { formatAsset } from "@webstudio-is/asset-uploader/server";
 
 export const loadByTreeId = async (treeId: Tree["id"]) => {
   const instancePropsEntries = await prisma.instanceProps.findMany({
     where: { treeId },
   });
 
+  // Fin all assetId in all props
+  const assetIds: string[] = [];
+
+  for (const instanceProps of instancePropsEntries) {
+    const props = JSON.parse(instanceProps.props);
+    const dpProps = UserDbProps.parse(props);
+
+    for (const dbProp of dpProps) {
+      if (dbProp.assetId != null) {
+        assetIds.push(dbProp.assetId);
+      }
+    }
+  }
+  // Load all assets
+  const assets = await prisma.asset.findMany({
+    where: {
+      id: {
+        in: assetIds,
+      },
+    },
+  });
+
+  const assetsMap = new Map(
+    assets.map((asset) => [asset.id, formatAsset(asset)])
+  );
+
   return instancePropsEntries.map((instanceProps) => {
     const props = JSON.parse(instanceProps.props);
-    UserProps.parse(props);
+    const resProps: UserProp[] = [];
+    // Can be changed onto `props as ` having that above is already checked, no optmizations now
+    const dpProps = UserDbProps.parse(props);
+
+    for (const dbProp of dpProps) {
+      if (dbProp.assetId != null) {
```

Comment:
can we please keep the nesting level low?
```suggestion
      if (dbProp.assetId == null) {
         resultProps.push(parsedProp)
         continue
      }
```